### PR TITLE
Improve RemoteCQRS logs

### LINKED ISSUE #26: Improve logging in CQRS pipeline
Currently, the logs that `RemoteCQRS` + `CQRS` pipeline produce are somewhat contradictory (i.e. `info: command executed successfully` is displayed after `warning: command is not valid`). They impact debugability slightly and we should fix that.

### PR DESCRIPTION
**Pull request by Dragemil**
#26

### CODE REVIEW

File: src/Domain/LeanCode.CQRS.Execution/CommandExecutor.cs [hunk 1]
```diff
@@ -29,7 +29,11 @@ public async Task<CommandResult> RunAsync(TAppContext appContext, ICommand comma
             try
             {
                 var res = await executor.ExecuteAsync(appContext, command);
-                logger.Information("Command {@Command} executed successfully", command);
+                if (res.WasSuccessful)
```
Comment (lukaszgarstecki) [inline src/Domain/LeanCode.CQRS.Execution/CommandExecutor.cs]: Maybe some meaningful log for case when result is not successful? :thinking:

**Review by lukaszgarstecki — DISMISSED** [review]

**Comment by jakubfijalkowski** [comment]
What logs will be printed in cases like:
1) Malformed payload?
2) Invalid (i.e. validator rejects) payload?
3) Unauthorized?
4) Not authenticated?
5) Exception thrown inside pipeline?
6) Exception thrown in handler?
7) Success?

Some excerpts would be great.

File: src/Domain/LeanCode.CQRS.Execution/CommandExecutor.cs (hunk 1 above)
Comment (Dragemil) [reply src/Domain/LeanCode.CQRS.Execution/CommandExecutor.cs]: If the result is not succesful it means it failed validation, and there is a warning in the `ValidationElement` in the pipeline.

File: src/Domain/LeanCode.CQRS.Security/CQRSSecurityElement.cs [hunk 2]
```diff
@@ -29,6 +29,8 @@ public CQRSSecurityElement(IAuthorizerResolver<TAppContext> authorizerResolver)
 
             if (customAuthorizers.Count > 0 && !(user?.Identity?.IsAuthenticated ?? false))
             {
+                logger.Warning("Unauthenticated user requested object that requires authorization");
```
Comment (Saancreed) [inline src/Domain/LeanCode.CQRS.Security/CQRSSecurityElement.cs]: The exception thrown just below will result in an error logged just after this anyway.

**Comment by Dragemil** [comment]
Some snippets of logs for various cases for the most recent commit:

1. Malformed payload
```
[11:02:27 WRN] Cannot deserialize object body from the request stream for type <command>                                                                          
System.Text.Json.JsonException: Expected depth to be zero at the end of the JSON payload. There is an open JSON object or array that should be closed. Path: $.PushNotificationToken | LineNumber: 6 | BytePositionInLine: 0. 
```

2. Invalid (i.e. validator rejects) payload
```
[12:21:27 WRN] Command {"FirstName": "name", "LastName": "surname", "EmailAddress": "", "$type": "UpdateProfileData"} is not valid with result {"Errors": [{"PropertyName": "EmailAddress", "ErrorMessage": "'Email Address' must not be empty.", "ErrorCode": 5, "$type": "ValidationError"}, {"PropertyName": "EmailAddress", "ErrorMessage": "'Email Address' is not a valid email address.", "ErrorCode": 7, "$type": "ValidationError"}], "IsValid": false, "$type": "ValidationResult"}
```

3. Unauthorized
```
[12:18:20 WRN] User does not have sufficient permissions (["admin"]) to run {"FirstName": "name", "LastName": "surname", "EmailAddress": "email", "$type": "UpdateProfileData"}
[12:18:20 WRN] Authorizer LeanCode.CQRS.Security.DefaultPermissionAuthorizer failed to authorize the user to run {"FirstName": "name", "LastName": "surname", "EmailAddress": "email", "$type": "UpdateProfileData"} of type BreakAndWash.Core.Contracts.Mobile.Profile.UpdateProfileData
```

4. Not authenticated
```
[12:11:45 WRN] Unauthenticated user requested {"FirstName": "", "LastName": "", "EmailAddress": "", "$type": "UpdateProfileData"} of type <type>, which requires authorization
```

5. Exception thrown inside pipeline
```
[12:28:56 ERR] Cannot execute object {"FirstName": "name", "LastName": "surname", "EmailAddress": "email", "$type": "UpdateProfileData"} of type <type>
System.Exception: Exception of type 'System.Exception' was thrown.
```

6. Exception thrown in handler
```
[12:31:14 ERR] Cannot execute object {"FirstName": "name", "LastName": "surname", "EmailAddress": "email", "$type": "UpdateProfileData"} of type <type>
System.Exception: Exception of type 'System.Exception' was thrown.
```
(The stack trace is ofc different then in 5.)

7. Success
```
[12:34:17 INF] Command {"FirstName": "name", "LastName": "surname", "EmailAddress": "email", "$type": "UpdateProfileData"} executed successfully
[12:34:17 INF] Remote object of type <type> executed successfully
```

I think those are appropriate.

File: src/Domain/LeanCode.CQRS.RemoteHttp.Server/BaseRemoteObjectHandler.cs [hunk 3]
```diff
@@ -68,14 +68,25 @@ public async Task<ExecutionResult> ExecuteAsync(HttpContext context)
...
+                    obj,
+                    type);
             }
-            catch (InsufficientPermissionException)
+            catch (InsufficientPermissionException ex)
             {
                 result = ExecutionResult.Fail(StatusCodes.Status403Forbidden);
+
+                Logger.Warning(
+                    "Authorizer {Authorizer} failed to authorize the user to run {@Object} of type {Type}",
```
Comment (lukaszgarstecki) [inline src/Domain/LeanCode.CQRS.RemoteHttp.Server/BaseRemoteObjectHandler.cs:82]: Why don't you pass ex as first parameter?

Comment (Dragemil) [reply src/Domain/LeanCode.CQRS.RemoteHttp.Server/BaseRemoteObjectHandler.cs:82]: There's nothing of value in the exception itself, that isn't already in this log.

**Review by lukaszgarstecki — APPROVED** [review]

File: src/Domain/LeanCode.CQRS.Security/CQRSSecurityElement.cs (hunk 2 above)
Comment (Dragemil) [reply src/Domain/LeanCode.CQRS.Security/CQRSSecurityElement.cs]: Not a case anymore.

**Review by jakubfijalkowski — APPROVED** [review]

**Pull request merged**